pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -841,14 +926,35 @@ func permissionForKeyValues(key string, values []string) *policyproto.Permission
 		}
 	}
 
-	return &policyproto.Permission{Rule: orRules}
+	if len(orRules.OrRules.Rules) > 0 {
+		return &policyproto.Permission{Rule: orRules}
+	}
+	return nil
 }
 
 // Create a Principal based on the key and the value.
 // key: the key of a subject property.
 // value: the value of a subject property.
 // forTCPFilter: the principal is used in the TCP filter.
 func principalForKeyValue(key, value string, forTCPFilter bool) *policyproto.Principal {
+	// Generate an any rule to grant access permission to anyone if the value is "*" for
+	// |attrSrcPrincipal|.
+	if key == attrSrcPrincipal {
+		if value == allUsers {
+			return &policyproto.Principal{
+				Identifier: &policyproto.Principal_Any{
+					Any: true,
+				},
+			}
+		}
+		// We don't allow users to use "*" in names or not_names. However, we will use "*" internally to
+		// refer to authenticated users, since existing code using regex to map "*" to all authenticated
+		// users.
+		if value == allAuthenticatedUsers {
+			value = "*"
```

Comment:
I'm not sure I understand your question completely. 

We don't allow "*" in the `names` or `not_names` field from a user perspective. https://github.com/istio/istio/blob/aa8763086c4958bea230485b65da824531ce05ab/pilot/pkg/model/validation.go#L1605-L1612

We only allow the value of `attrSrcPrincipal` to be `allAuthenticatedUsers` or `allUsers`. For `allUsers`, we set `principal_any` to `true`.
https://github.com/istio/istio/blob/aa8763086c4958bea230485b65da824531ce05ab/pilot/pkg/networking/plugin/authz/rbac.go#L943-L949

Reply:
You set `value` to `*` in line 954, how would this propagate to envoy filter config and how does it deal with the un-authenticated and authenticated user?

Reply:
I had some comment above that line of code to discuss it.

> We don't allow users to use * in names or not_names. However, we will use * internally to refer to authenticated users, since existing code using regex to map * to all authenticated users.

So `allUsers` will create a Principal_Any, `allAuthenticatedUsers` will create a Principal_Authenticated

Reply:
I was trying to ask how does it filter out those un-authenticated users in envoy. See my comment in the unit test, also it seems it's not using Principal_Authenticated but actually the metadata string matcher.